src/fides/api/ctl/migrations/versions/eb1e6ec39b83_add_role_based_permissions.py

```diff
@@ -0,0 +1,63 @@
+"""add role based permissions
+
+Revision ID: eb1e6ec39b83
+Revises: d65bbc647083
+Create Date: 2023-02-24 19:27:17.844231
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+from sqlalchemy.dialects import postgresql
+
+revision = "eb1e6ec39b83"
+down_revision = "d65bbc647083"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column(
+        "client",
+        sa.Column("roles", sa.ARRAY(sa.String()), server_default="{}", nullable=False),
```

Comment:
thinking about the enum discussion, is there a reason to not use an enum in the db? I guess it'd require a migration when adding/deleting a role which is a pain, but arguably doing adding/deleting a role may already require a data migration of some sort for existing users. 

I ask because when I was [playing around with enums for scopes](https://github.com/ethyca/fides/discussions/2642#discussioncomment-5092052), I couldn't make the `get_user_permissions` func/endpoint typed to return a list of the Scope enums since it wasn't an enum in the db. I was able to work around it by instead having `read_scopes` return the list of Scope enums, since for my use case the frontend just needed an up to date list of the scopes. but this actually results in some type fudging that needs to be done on the frontend to convince typescript that `get_user_permissions`'s scopes are the same type of scopes that `read_scopes` returns. the type fudging is really easy to do, but is technically unsafe, and I wonder if that might mean it's a lil unsafe in other places too. 

migrating to an enum for scopes in the db seems like a big pain as well 😅 but mostly just want to get the discussion going, even if we choose not to go that route!

Reply:
@allisonking can you show me a snippet of what you'd need the API response to look like?  I feel like there's a way to change the format of `get_user_permissions` to be what you need without making this be a database migration - I can look into this further I'm hesitant to add a db migration for this one since every time a new endpoint goes in we likely need new scopes.

Reply:
yup yup!